Improve wording around Secrets as environment variables (#38083)
kubernetes/website: content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md

path of each key:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  containers:
  - name: mypod
    image: redis
    volumeMounts:
    - name: foo
      mountPath: "/etc/foo"
      readOnly: true
  volumes:
  - name: foo
    secret:
      secretName: mysecret
      items:
      - key: username
        path: my-group/my-username
```

When you deploy this Pod, the following happens:

* The `username` key from `mysecret` is available to the container at the path
  `/etc/foo/my-group/my-username` instead of at `/etc/foo/username`.
* The `password` key from that Secret object is not projected.

If you list keys explicitly using `.spec.volumes[].secret.items`, consider the
following:

* Only keys specified in `items` are projected.
* To consume all keys from the Secret, all of them must be listed in the
  `items` field.
* All listed keys must exist in the corresponding Secret. Otherwise, the volume
  is not created.

### Set POSIX permissions for Secret keys

You can set the POSIX file access permission bits for a single Secret key.
If you don't specify any permissions, `0644` is used by default.

> You can also set a default mode for the entire Secret volume and override per key if needed.

Reviewer: ```suggestion
You can also set a default POSIX file mode for the entire Secret volume, and you can 
override that default per key if needed.
```

For example, you can specify a default mode like this:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: mypod
spec:
  containers:
  - name: mypod
    image: redis
    volumeMounts:
    - name: foo
      mountPath: "/etc/foo"
  volumes:
  - name: foo
    secret:
      secretName: mysecret
      defaultMode: 0400
```

> The secret is mounted on `/etc/foo`; all the files created by the

Reviewer: ```suggestion
The Secret is mounted on `/etc/foo`; all the files created by the
```

secret volume mount have permission `0400`.

{{< note >}}
If you're defining a Pod or a Pod template using JSON, beware that the JSON

> specification doesn't support octal notation. You can use the decimal value
> for the `defaultMode` (for example, 0400 in octal is 256 in decimal) instead.

Reviewer: ```suggestion
specification doesn't support octal literals for numbers; JSON sees `0400` as the _decimal_ value
400, when in fact `0400` in octal is 256 in decimal. In JSON, use the decimal value for the
`defaultMode` instead.

```
